File: solutions/clojure/pascals-triangle/1/src/pascals_triangle.clj
(ns pascals-triangle)


(def binom 
  (memoize (fn  [n k] 
             (condp = [n k] 
               [0 0] 1 
               (if (or (= k 0) (= k n)) 1  
                 (bigint 
                  (+ (binom (dec n) k) (binom (dec n) (dec k)))))))))

              


(defn row [num] ;; <- arglist goes here
  ;; your code goes here
  (for [k (range 0 num)] (binom (dec num) k))
)

(def triangle
  (map #(row (inc %)) (range) )
  )

File: solutions/julia/pascals-triangle/4/pascals-triangle.jl
function triangle(n)
    n>=0 || throw(DomainError(n))
    n==0 && return []
    n==1 && return [[1]]

    current = [[1],[1,1]]
    for row in 3:n
        prev = current[end]
        push!(current, append!([1],[prev[i]+prev[i+1] for i in 1:(length(prev)-1)],[1]))
    end
    current
end